components/network/MessagingAppsChart.js

```diff
@@ -0,0 +1,133 @@
+import React, { useMemo } from 'react'
+import { useRouter } from 'next/router'
+import { Heading, Box, Flex } from 'ooni-components'
+import axios from 'axios'
+
+import { axiosResponseTime } from 'components/axios-plugins'
+import useSWR from 'swr'
+
+import GridChart, { prepareDataForGridChart } from 'components/aggregation/mat/GridChart'
+import { DetailsBox } from 'components/measurement/DetailsBox'
+
+const swrOptions = {
+  revalidateOnFocus: false,
+  dedupingInterval: 10 * 60 * 1000,
+}
+
+const baseURL = process.env.NEXT_PUBLIC_AGGREGATION_API
+axiosResponseTime(axios)
+
+const messagingFetcher = (query) => {
+  const reqUrl = `${baseURL}/api/v1/aggregation?${query}`
+  const result = []
+
+  const request = (reqUrl) => { 
+    return axios.get(reqUrl).then(r => {
+      if (!r?.data?.result) {
+        const error = new Error(`Request ${reqUrl} did not contain expected result`)
+        error.data = r
+        throw error
+      }
+      return {
+        data: r.data.result,
+        loadTime: r.loadTime,
+        url: r.config.url
+      }
+    }).catch(e => {
+      console.log(e)
+      e.message = e?.request?.response ?? e.message
+      throw e
+    })
+  }
+
+  const signal = request(`${reqUrl}&test_name=signal`)
+  const telegram = request(`${reqUrl}&test_name=telegram`)
+  const whatsapp = request(`${reqUrl}&test_name=whatsapp`)
+  const fbMessenger = request(`${reqUrl}&test_name=facebook_messenger`)
```

Comment:
This is a good interim solution for this, though we ought to really be able to do this with a single request to the backend, see: https://github.com/ooni/backend/issues/584